clang/lib/DPCT/CommandOption/DpctOptions.cpp

```diff
@@ -174,6 +174,8 @@ void DpctOptionBase::init() {
                 DpctOptionNameKind::OPT_InRoot);
   setDependency(DpctOptionNameKind::OPT_BuildScriptFile,
                 DpctOptionNameKind::OPT_GenBuildScript);
+  setDependency(DpctOptionNameKind::OPT_MigrateBuildScriptOnly,
```

Comment:
if we set --migrate-build-script and no build script kind set, both cmake and python script need to be migrated. So no dependency here.

Reply:
By the new design, no migration happens when only **migrate-build-script-only** is specified.
This option needs **migrate-build-script** to specify which build script types (CMake or Python) to migrate.

So, I setup a one way dependency i.e., **migrate-build-script-only** depends on **migrate-build-script**

Reply:
Reverted the change now